### DISCUSSION TITLE
Online Decoding

### DISCUSSION THREAD

**Discussion by ErnstTmp** [espnet/espnet#3608] [Q&A] [answered]
Hello!

I would like to use espnet for Online Decoding. Is this possible - if yes, could you advise me ho to start, if not, will this be made possible in future? 
Online decoding is the only feature that is missing for me. I could not find something for online decoding and espnet so
I am posting it here.
Thanks a lot,
Ernst

**Comment by sw005320** [comment, marked as answer]
We actually have several ongoing projects.
The readiest one is to use http://arxiv-export-lb.library.cornell.edu/abs/2006.14941.
The basic implementation is already in the master.
Keqi is working on providing an example based on @hfujihara's fork.

**Comment by D-Keqi** [comment]
@ErnstTmp Hi! Online decoding is an ongoing preject currently and we will make a PR as soon as possible.
And we have an example here https://github.com/D-Keqi/espnet/tree/dev/real_streaming_decoder/egs2/aishell/asr1
Please feel free to ask me any questions!

**Comment by ErnstTmp** [comment]
@D-Keqi @sw005320 Hi! Thank you very much for your answers. I was away some time due to medical reasons, but now I am back and I am trying to run the example from Keqi with the German Common Voice dataset. It looked quote good, but ASR-Training stops with. the messages **2021-10-25 21:37:38,741 (trainer:582) WARNING: The grad norm is nan. Skipping updating the model.** I tried to debug, but the wav files seem to be correct and I could not find any error. Thank you very much for your help!!

**Reply by D-Keqi** [reply]
https://github.com/espnet/espnet/issues/3185#issuecomment-829644185
I think this may help you.

**Comment by sw005320** [comment]
We've made a demo
https://github.com/espnet/notebook/blob/master/espnet2_streaming_asr_demo.ipynb
This is only for Mandarin, but @D-Keqi will add English and Japanese models.
We may need more documentation and we welcome any input.
